core/src/main/java/org/apache/calcite/rel/rules/DateRangeRules.java

```diff
@@ -401,21 +402,33 @@ private RexNode toRex(RexNode operand, Range<Calendar> r) {
             : SqlStdOperatorTable.GREATER_THAN;
         nodes.add(
             rexBuilder.makeCall(op, operand,
-                rexBuilder.makeDateLiteral(
-                    DateString.fromCalendarFields(r.lowerEndpoint()))));
+                makeLiteralForComparison(rexBuilder, r.lowerEndpoint(), operand)));
       }
       if (r.hasUpperBound()) {
         final SqlBinaryOperator op = r.upperBoundType() == BoundType.CLOSED
             ? SqlStdOperatorTable.LESS_THAN_OR_EQUAL
             : SqlStdOperatorTable.LESS_THAN;
         nodes.add(
             rexBuilder.makeCall(op, operand,
-                rexBuilder.makeDateLiteral(
-                    DateString.fromCalendarFields(r.upperEndpoint()))));
+                makeLiteralForComparison(rexBuilder, r.upperEndpoint(), operand)));
```

Comment:
can we name this in a way it reflects that is creating Dates and Timestamp literals?

Reply:
Agreed. Will rename to "dateTimeLiteral".